vizro-ai/src/vizro_ai/dashboard/_response_models/controls.py

```diff
@@ -100,31 +92,58 @@ class ControlPlan(BaseModel):
         to control a specific component, include the relevant component details.
         """,
     )
-    df_name: str = Field(
+    target_components_id: List[str] = Field(
         ...,
         description="""
-        The name of the dataframe that the target component will use.
-        If the dataframe is not used, please specify that.
+        The id of the target components that this control will affect.
         """,
     )
 
-    def create(self, model, controllable_components, all_df_metadata) -> Optional[vm.Filter]:
+    def _get_target_df_name(self, components_plan, controllable_components):
+        target_controllable = set(self.target_components_id) & set(controllable_components)
+        df_names = {
+            component_plan.df_name
+            for component_plan in components_plan
+            if component_plan.component_id in target_controllable
+        }
+
+        if len(df_names) > 1:
+            logger.warning(
+                f"""
+[FALLBACK] Multiple dataframes found in the target components: {df_names}.
+Choose one dataframe to build the filter.
+"""
```

Comment:
I think I do not understand this. Should the logic not be:
- model determines which components should be controlled
- we then check if those components can be controlled, and take a subset at worst
- the remaining dfs are what we get the metadata for (and the could be multiple no?)